analyzers/tests/SonarAnalyzer.UnitTest/TestCases/UninvokedEventDeclaration.cs

```diff
@@ -226,4 +226,35 @@ public void SomeMethod()
             }
         }
     }
+
+    public class EventChainingDirect
+    {
+        public event EventHandler Logging; // Noncompliant - FP
+
+        public void RunTest()
+        {
+            EventTestBase eventTestBase = new EventTestBase();
+            eventTestBase.Logging += Logging;
+            eventTestBase.RunSomethingThatSendAnEvent();
+        }
+    }
+
+    public class EventChainingLambda
+    {
+        public event EventHandler Logging;
+
+        public void RunTest()
+        {
+            EventTestBase eventTestBase = new EventTestBase();
+            eventTestBase.Logging += (s, e) => Logging(s, e);
+            eventTestBase.RunSomethingThatSendAnEvent();
+        }
+    }
+
+    public class EventTestBase
```

Comment:
could you please move this above `EventChainingDirect`? because I was confused if `EventChainingLambda` had something to do with the repro or not...

But I understand now that `EventChainingDirect` and `EventChainingLambda` are just to separate use cases with `EventTestBase`, and they don't interact with each other.

Thanks!